### PR TITLE
feat(mkdocs): add automatic documentation using mkdocs

### LINKED ISSUE #16: Automatic Documentation
Create and maintain a documentation page that would be updated using GitHub actions based on docstrings.

### PR DESCRIPTION
**Pull request by SauravMaheshkar**
Fixes #16.

Ideally this should be worked on after #13 is merged. Following that let's rebase and then proceed with adding docstrings & typehints to `restorers/`.

### CODE REVIEW

**Comment by SauravMaheshkar** [comment]
@soumik12345 If you could checkout to this branch and run

```lang-bash
mkdocs serve
```

**Comment by soumik12345** [comment]
> @soumik12345 If you could checkout to this branch and run
> 
> ```
> mkdocs serve
> ```

Hi @SauravMaheshkar I am getting the following error:

```
Error: Config file 'mkdocs.yml' does not exist.
```

**Comment by ariG23498** [comment]
Wouldn't it be better if we could review the documentation on the website (restorer docs) rather than here?

**Comment by soumik12345** [comment]
> Wouldn't it be better if we could review the documentation on the website (restorer docs) rather than here?

**Comment by soumik12345** [comment]
> Wouldn't it be better if we could review the documentation on the website (restorer docs) rather than here?

@ariG23498 you can, just hit

```bash
pip install -q mkdocs mkdocstrings mkdocstrings-python
mkdocs serve
```

File: restorers/model/nafnet/nafblock.py [hunk 1]
```diff
@@ -99,32 +129,59 @@ def get_config(self) -> dict:
...
+
+        Reference:
+
+        1. [Simple Baselines for Image Restoration](https://arxiv.org/abs/2204.04676)
+
+        Parameters:
+    <<<<<<< HEAD
+    =======
+            input_channels (Optional[int]): number of channels in the input (as NAFBlock retains the input size in the output)
+    >>>>>>> ea3089f6eb211c37a2a28eb183312fd8a0c1d106
```
Comment (ariG23498) [inline restorers/model/nafnet/nafblock.py]: I think this is a merge conflict @soumik12345

Comment (soumik12345) [reply restorers/model/nafnet/nafblock.py]: cc: @Samarendra109

**Pull request closed without merging**